webapp/channels/src/components/cloud_preview_modal/preview_modal_controller.tsx

```diff
@@ -0,0 +1,137 @@
+                >
+                    {contentData.map((_, index) => (
+                        <div
+                            key={`dot-${index}`}
+                            className={`preview-modal-controller__dot ${index === currentIndex ? 'preview-modal-controller__dot--active' : ''}`}
+                        />
+                    ))}
+                </div>
+                <span className='preview-modal-controller__page-counter'>
+                    {currentIndex + 1}{'/'}{contentData.length}
+                </span>
+            </div>
+
+            <div className='preview-modal-controller__navigation-buttons'>
+                {isFirstSlide && (
+                    <button
+                        className='preview-modal-controller__skip-button'
+                        onClick={handleSkip}
+                    >
+                        <FormattedMessage
+                            id='cloud_preview_modal.skip'
+                            defaultMessage='Skip for now'
+                        />
+                    </button>
+                )}
+
+                {!isFirstSlide && (
+                    <button
+                        className='preview-modal-controller__nav-button preview-modal-controller__nav-button--tertiary'
+                        onClick={handlePrevious}
+                    >
+                        <ChevronLeftIcon size={18}/>
+                        <FormattedMessage
+                            id='cloud_preview_modal.previous'
+                            defaultMessage='Previous'
+                        />
+                    </button>
+                )}
+
+                <button
+                    className='preview-modal-controller__nav-button preview-modal-controller__nav-button--primary'
+                    onClick={isLastSlide ? onClose : handleNext}
+                >
+                    {isLastSlide ? (
+                        <FormattedMessage
+                            id='cloud_preview_modal.done'
+                            defaultMessage='Finish'
+                        />
+                    ) : (
+                        <>
+                            <FormattedMessage
+                                id='cloud_preview_modal.next'
+                                defaultMessage='Next'
+                            />
+                            <ChevronRightIcon size={18}/>
+                        </>
+                    )}
+                </button>
+            </div>
+        </div>
+    );
+
+    return (
+        <GenericModal
+            show={show}
+            onHide={onClose}
+            compassDesign={true}
+            enforceFocus={true}
+            ariaLabel='Cloud Preview Introduction'
+            showHeader={false}
+            showCloseButton={false}
+            bodyPadding={true}
+            footerContent={footerContent}
+            className='preview-modal-controller'
+        >
+            <button
```

Comment:
Couldn't this use the built-in close button in the modal?

Reply:
That sounds like that might be an issue with the design then since the modal content has a title, so why aren't we putting that into the modal header? Looking at the screenshot above, I see that the SKU badge is above the title, so perhaps it could go below it?

This feels like a case where the design isn't following the standard pattern for modals which is leading to us having to jump through hoops with the code as well

CC @matthewbirtch